From 774c113f586273125c359dabe23bbefa055fb998 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@st.com>
Date: Fri, 21 Oct 2022 12:09:30 +0200
Subject: [PATCH 053/954] feat(stm32mp15-fdts): apply RCC security

RCC is secured when "st,stm32mp1-rcc-secure" compatible is used in DT.
Force this new compatible for ST boards.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git f84fdb930676046baf8e2a6f9c009d7204d1324a feat(stm32mp15-fdts): apply RCC security]
Change-Id: I1b51344c19bf2e21d011e185874fbb98ba1c987a
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
---
 fdts/stm32mp157a-ed1.dts  | 1 +
 fdts/stm32mp157c-ed1.dts  | 1 +
 fdts/stm32mp157d-ed1.dts  | 1 +
 fdts/stm32mp157f-ed1.dts  | 1 +
 fdts/stm32mp15xx-dkx.dtsi | 1 +
 5 files changed, 5 insertions(+)

diff --git a/fdts/stm32mp157a-ed1.dts b/fdts/stm32mp157a-ed1.dts
index bdb5604185..cab39b08f8 100644
--- a/fdts/stm32mp157a-ed1.dts
+++ b/fdts/stm32mp157a-ed1.dts
@@ -186,6 +186,7 @@
 };
 
 &rcc {
+	compatible = "st,stm32mp1-rcc-secure", "st,stm32mp1-rcc", "syscon";
 	st,clksrc = <
 		CLK_MPU_PLL1P
 		CLK_AXI_PLL2P
diff --git a/fdts/stm32mp157c-ed1.dts b/fdts/stm32mp157c-ed1.dts
index 949c929af5..5c3acec1cf 100644
--- a/fdts/stm32mp157c-ed1.dts
+++ b/fdts/stm32mp157c-ed1.dts
@@ -190,6 +190,7 @@
 };
 
 &rcc {
+	compatible = "st,stm32mp1-rcc-secure", "st,stm32mp1-rcc", "syscon";
 	st,clksrc = <
 		CLK_MPU_PLL1P
 		CLK_AXI_PLL2P
diff --git a/fdts/stm32mp157d-ed1.dts b/fdts/stm32mp157d-ed1.dts
index 454452c489..9d2a6860eb 100644
--- a/fdts/stm32mp157d-ed1.dts
+++ b/fdts/stm32mp157d-ed1.dts
@@ -186,6 +186,7 @@
 };
 
 &rcc {
+	compatible = "st,stm32mp1-rcc-secure", "st,stm32mp1-rcc", "syscon";
 	st,clksrc = <
 		CLK_MPU_PLL1P
 		CLK_AXI_PLL2P
diff --git a/fdts/stm32mp157f-ed1.dts b/fdts/stm32mp157f-ed1.dts
index bab448c7e0..8c69b4b872 100644
--- a/fdts/stm32mp157f-ed1.dts
+++ b/fdts/stm32mp157f-ed1.dts
@@ -190,6 +190,7 @@
 };
 
 &rcc {
+	compatible = "st,stm32mp1-rcc-secure", "st,stm32mp1-rcc", "syscon";
 	st,clksrc = <
 		CLK_MPU_PLL1P
 		CLK_AXI_PLL2P
diff --git a/fdts/stm32mp15xx-dkx.dtsi b/fdts/stm32mp15xx-dkx.dtsi
index f8baa9d4a6..101063b1e9 100644
--- a/fdts/stm32mp15xx-dkx.dtsi
+++ b/fdts/stm32mp15xx-dkx.dtsi
@@ -194,6 +194,7 @@
 };
 
 &rcc {
+	compatible = "st,stm32mp1-rcc-secure", "st,stm32mp1-rcc", "syscon";
 	st,clksrc = <
 		CLK_MPU_PLL1P
 		CLK_AXI_PLL2P
-- 
2.53.0

